### PR TITLE
Add http fetcher using req

### PR DESCRIPTION
**Pull request by louiscb**

### CODE REVIEW

**Comment by louiscb** [comment]
Tested in PLUR with this file and could successfully get the term by calling `AppAudienceServer.Auctions.VendorList.get()`

```
defmodule AppAudienceServer.Auctions.VendorList do
  use RemotePersistentTerm

  @fetcher RemotePersistentTerm.Fetcher.Http

  def child_spec(_) do
    %{
      id: __MODULE__,
      start: {
        __MODULE__,
        :start_link,
        [
          [
            fetcher_mod: @fetcher,
            fetcher_opts: [url: "https://vendor-list.consensu.org/v3/vendor-list.json"],
            refresh_interval: :timer.minutes(1),
            lazy_init?: true
          ]
        ]
      }
    }
  end

end
```

**Comment by sneako** [comment]
Please add some tests

File: lib/remote_persistent_term/fetcher/http.ex [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  def current_version(_state) do
+    {:ok, DateTime.utc_now() |> DateTime.to_string()}
+  end
+
+  @impl true
+  def download(state) do
+    Logger.info("downloading remote term from #{state.url}")
+
+    with {:ok, resp} <- Req.get(state.url, cache: true) do
+      if resp.status < 300 do
```
Comment (sneako) [inline lib/remote_persistent_term/fetcher/http.ex:49]: 304 is the status code for "Not Modified", are you sure we want to return an error in that case?

Comment (louiscb) [reply lib/remote_persistent_term/fetcher/http.ex:49]: We wouldn't want to replace the term with the contents of that 304 body, which would be null. I would assume that Req caching would handle that for us and return a 200. I'll test that locally to see 

https://github.com/wojtekmach/req/blob/50c7d3ee2ceea2c64f88526efb49a81ec173cd9c/lib/req/steps.ex#L557

Comment (louiscb) [reply lib/remote_persistent_term/fetcher/http.ex:49]: yes req returns a 200 from a 304

Comment (sneako) [reply lib/remote_persistent_term/fetcher/http.ex:49]: ah cool, lets add a test case that shows that. We can use [bypass](https://github.com/PSPDFKit-labs/bypass) for example

Comment (louiscb) [reply lib/remote_persistent_term/fetcher/http.ex:49]: cool, good idea

**Review by sneako — APPROVED** [review]

**Pull request merged**